library/agent/wrapRequire.ts

```diff
@@ -0,0 +1,61 @@
+import * as mod from "module";
+import { ModifyingRequireInterceptor } from "./hooks/ModifyingRequireInterceptor";
+import { defineProperty } from "../helpers/wrap";
+import { Agent } from "./Agent";
+const req = mod.prototype.require;
+
+const interceptors: ModifyingRequireInterceptor[] = [];
+
+export function wrapRequire(agent: Agent) {
+  // @ts-expect-error Ignore type error
+  mod.prototype.require = function wrap(id) {
+    const interceptor = interceptors.find((i) => i.getName() === id);
+    if (!interceptor) {
+      return req.apply(this, [id]);
+    }
+
+    const original = req.apply(this, [id]);
+
+    if (typeof original !== "function") {
+      throw new Error(
+        "Module must export a function to be wrapped during require"
+      );
+    }
+
+    if (original.__wrapped) {
```

Comment:
Would prefer to keep this in state, which modules have been patched? Instead of setting __wrapped?